src/app-layer-ftp.c

```diff
@@ -642,14 +642,18 @@ static AppLayerResult FTPParseRequest(Flow *f, void *ftp_state,
                      * Min size has been checked in FTPParseRequestCommand
                      * PATH_MAX includes the null
                      */
-                    int file_name_len = MIN(PATH_MAX - 1, state->current_line_len - 5);
+                    uint32_t file_name_len = MIN(PATH_MAX - 1, state->current_line_len - 5);
+                    if (file_name_len > UINT16_MAX) {
```

Comment:
this is a runtime check for something that shouldn't be possible in practice, as iirc PATH_MAX is 4k. Should we use a static assertion here?
https://riptutorial.com/c/example/1842/static-assertion

Reply:
to clarify: if we have a static assert that PATH_MAX < UINT16_MAX we can get rid of a runtime check like this

Reply:
Indeed, will do

Reply:
What about 
```
#if PATH_MAX > UINT16_MAX
#error PATH_MAX is greater than UINT16_MAX
#endif
```
I did not know about static assertions

Reply:
why not use the thing that is designed to do this, meaning static assertions?

Reply:
> why not use the thing that is designed to do this, meaning static assertions?

For what I knew `#error` is designed to do this, I did not know about static assertions

They seem to be equivalent here :
https://wiki.sei.cmu.edu/confluence/display/c/DCL03-C.+Use+a+static+assertion+to+test+the+value+of+a+constant+expression

`static_assert` style looks more compact.
So, it is in branch v4.
Are there other comments or should I make a PR with v4 ?